src/pixelator/plot/spatial_analysis_plots.py

```diff
@@ -101,102 +108,100 @@ def plot_colocalization_heatmap(
     return plt.gcf(), plt.gca()
 
 
-def _get_top_marker_pairs(
-    colocalization_data: pd.DataFrame,
-    n_top_marker_pairs: int,
-    value_col: str = "pearson",
-) -> list:
-    colocalization_data["abs_val"] = colocalization_data[value_col].abs()
-    top_marker_pairs = colocalization_data.nlargest(n_top_marker_pairs, "abs_val")
-    top_markers = list(
-        set(top_marker_pairs["marker_1"]).union(set(top_marker_pairs["marker_2"]))
-    )
-
-    return top_markers
-
-
 def plot_colocalization_diff_heatmap(
     colocalization_data: pd.DataFrame,
-    target: str,
     reference: str,
+    targets: str | list[str] | None = None,
     contrast_column: str = "sample",
-    markers: Union[list, None] = None,
-    n_top_marker_pairs: Union[int, None] = None,
+    top_marker_log_p: float | None = None,
+    min_log_p: float = 3.0,
     cmap: str = "vlag",
     use_z_score: bool = True,
-) -> Tuple[plt.Figure, plt.Axes]:
+) -> Tuple[dict, dict]:
     """Plot the differential colocalization between reference and target components.
 
-    Example usage: plot_colocalization_diff_heatmap(pxl.colocalization, target:"stimulated", reference:"control", contrast_column="sample").
+    Example usage: plot_colocalization_diff_heatmap(pxl.colocalization,
+    reference:"control", contrast_column="sample").
 
-    :param colocalization_data: The colocalization data frame that can be found in a pixel variable "pxl" through pxl.colocalization. The data frame should contain the columns "marker_1", "marker_2", "pearson", "pearson_z", and the contrast_column.
-    :param target: The label for target components in the contrast_column.
+    :param colocalization_data: The colocalization data frame that can be found
+    in a pixel variable "pxl" through pxl.colocalization. The data frame should
+    contain the columns "marker_1", "marker_2", "pearson", "pearson_z", and the
+    contrast_column.
     :param reference: The label for reference components in the contrast_column.
+    :param targets: label or list of labels for target components in the
+    contrast_column. When not specified, all labels in the contrast_column
+    except the reference label are used as targets.
     :param contrast_column: The column to use for the contrast. Defaults to "sample".
-    :param markers: The markers to include in the heatmap. Defaults to None. At most only one of n_top_marker_pairs or markers should be provided.
-    :param n_top_marker_pairs: The number of top marker pairs to include in the heatmap. Defaults to None. At most only one of n_top_marker_pairs or markers should be provided.
+    :param top_marker_log_p: When set to a value, only markers with at least
+    one higher log10 p-score are plotted.
```

Comment:
Not sure if I understand this. Only markers with log10 p-value > `top_marker_log_p` are plotted?

Reply:
Yes, markers with differential colocalization with log10 p-value > top_marker_log_p with at least one other marker since the p-values are assigned to marker-pairs. I will rephrase it to be more explicit.